pkg/cfn/manager/tasks.go

```diff
@@ -0,0 +1,54 @@
+package manager
+
+import (
+	"sync"
+
+	"github.com/kubicorn/kubicorn/pkg/logger"
+)
+
+type task func(chan error) error
+
+// Run a set of tests in paralle and wait for them to complete;
+// passError should take any errors and do what it needs to in
+// a given context, e.g. duing serial CLI-driven execution one
+// can keep errors in a slice, while in a deamon channel maybe
+// more suitable
+func Run(passError func(error), tasks ...task) {
+	wg := &sync.WaitGroup{}
+	wg.Add(len(tasks))
+	for t := range tasks {
+		go func(t int) {
+			defer wg.Done()
+			logger.Debug("task %d started", t)
+			errs := make(chan error)
+			if err := tasks[t](errs); err != nil {
+				passError(err)
+				return
+			}
+			if err := <-errs; err != nil {
+				passError(err)
+				return
+			}
+			logger.Debug("task %d returned without errors", t)
+		}(t)
+	}
+	logger.Debug("waiting for %d tasks to complete", len(tasks))
+	wg.Wait()
+}
+
+// CreateClusterWithInitialNodeGroup runs two tasks to create
+// the stacks for use with CLI; any errors will be returned
+// as a slice on completion of one of the two taks
+func (s *StackCollection) CreateClusterWithInitialNodeGroup() []error {
```

Comment:
Would be good to have this visualised / documented in the wiki for when someone new to eksctl comes along. They'd know that that there are 2 stacks and the resources in each are x, y, z.

Reply:
Yes, indeed. I also think there are quite a few things that need to documented very specifically, i.e. why this is being done (even if it's just a pointer to the discussion in one the issues).